src/SqlLoader.ts

```diff
@@ -0,0 +1,352 @@
+    if (rows.length === 0) {
+      throw new Error(`No VCF Record returned for id ${id}`);
+    }
+    const mapped = mapRows(rows, meta, this.getCategories(), nestedTables);
+    if (mapped.length > 1) {
+      throw new Error(`More than 1 VCF Record returned for id ${id}`);
+    }
+    return mapped[0]!.data;
+  }
+
+  getCategories(): Categories {
+    if (this.categories === undefined) {
+      const sql = "SELECT field, id, value FROM categories";
+      const rows: SqlRow[] = executeSql(this.db as Database, sql, {});
+      const result = new Map<string, FieldCategories>();
+
+      for (const row of rows) {
+        const field = row.field as string;
+        const value = row.value as string;
+
+        let valueMap = result.get(field);
+        if (!valueMap) {
+          valueMap = new Map<number, string>();
+          result.set(field, valueMap);
+        }
+        valueMap.set(row.id as number, value);
+      }
+      this.categories = result;
+    }
+    return this.categories;
+  }
+
+  countMatchingSamples(query: Query | undefined): TableSize {
+    const categories = this.getCategories();
+    const { partialStatement, values } =
+      query !== undefined ? simpleQueryToSql(query, categories, {}) : { partialStatement: "", values: {} };
+    const whereClause = query !== undefined ? `WHERE ${partialStatement}` : "";
+    const sql = `SELECT 
+                        (SELECT COUNT(*) FROM sample) AS total_size, 
+                        COUNT(DISTINCT id) AS count
+                     from sample ${whereClause}`;
+    const rows = executeSql(this.db as Database, sql, values);
+    if (!rows || rows.length === 0 || rows[0] === undefined) return { size: 0, totalSize: 0 };
+    return { size: (rows[0]["count"] ?? 0) as number, totalSize: (rows[0]["total_size"] ?? 0) as number };
+  }
+
+  countMatchingPhenotypes(query: Query | undefined): TableSize {
+    const categories = this.getCategories();
+    const { partialStatement, values } =
+      query !== undefined ? simpleQueryToSql(query, categories, {}) : { partialStatement: "", values: {} };
+    const whereClause = query !== undefined ? `WHERE ${partialStatement}` : "";
+    const sql = `SELECT (SELECT COUNT(*) FROM phenotype) AS total_size,
+                            COUNT(DISTINCT p.id) AS count
+                                 FROM samplePhenotype sp
+                                          JOIN phenotype p ON sp.phenotype_id = p.id
+                                          JOIN sample sample ON sp.sample_id = sample.id
+                                     ${whereClause}`;
+    const rows = executeSql(this.db as Database, sql, values);
+    if (!rows || rows.length === 0 || rows[0] === undefined) return { size: 0, totalSize: 0 };
+    return { size: (rows[0]["count"] ?? 0) as number, totalSize: (rows[0]["total_size"] ?? 0) as number };
+  }
+}
+
+function base64ToUint8Array(): Uint8Array {
+  const binaryString = atob(SQLBASE64);
```

Comment:
don't use `atob`. from docs:
```
This function is only provided for compatibility with legacy web platform APIs and should never be used in new code, because they use strings to represent binary data and predate the introduction of typed arrays in JavaScript. For code running using Node.js APIs, converting between base64-encoded strings and binary data should be performed using Buffer.from(str, 'base64') and buf.toString('base64').
```
first see comment in `sqLiteBase64.ts`.